build.sc

```diff
@@ -1254,7 +1259,7 @@ object dev extends MillPublishScalaModule {
   )
 
   lazy val allPublishModules = build.millInternal.modules.collect {
-    case m: PublishModule if m ne this => m
+    case m: PublishModule if (m ne this) && (m ne dist) => m
```

Comment:
Can we change the name of that target or add some comment to clarify it's purpose?